lib/rules/prefer-name-spaced-react.js

```diff
@@ -0,0 +1,34 @@
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prefer React import be name-spaced.',
+      category: 'Stylistic Issues',
+      recommended: false,
+      uri:
+        'https://github.com/Shopify/eslint-plugin-shopify/blob/master/docs/rules/prefer-name-spaced-react.md',
+    },
+    fixable: null,
+  },
+  create(context) {
+    function report(node) {
+      const {name} = node;
+
+      context.report({
+        node,
+        message: `Prefer name-spaced React import. (import * as React from 'react';).`,
```

Comment:
I think we actually want the opposite here.

Reply:
I am curious whether TS will complain about the namespace import for React with that flag applied. My gut is that a namespace import is still more correct for React. In express, it made a lot less sense, because the namespace import got you a function, where it would always give you an object in “true” modules. React, in contrast, uses module.exports as an object that just exposes named properties, so a namespace import is more appropriate.